src/java/main/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -267,10 +267,17 @@ private boolean checkFourLetterWord(final Channel channel,
     {
         // We take advantage of the limited size of the length to look
         // for cmds. They are all 4-bytes which fits inside of an int
-        String cmd = FourLetterCommands.getCmdMapView().get(len);
-        if (cmd == null) {
+        if (!FourLetterCommands.isKnown(len)) {
             return false;
         }
+
+        // ZOOKEEPER-2693: don't execute 4lw if it's not enabled.
+        String cmd = FourLetterCommands.getCommandString(len);
+        if (!FourLetterCommands.isEnabled(cmd)) {
+            LOG.debug("Command {} is not executed because it is not white listed.", cmd);
+            return true;
```

Comment:
We can create a dummy command(ErrorCommand extends AbstractFourLetterCommand ) and execute it and return true same way as SetTraceMaskCommand 
This ErrorCommand will take care rest of the thing, sending any error message to client and closing the connection etc.